src/util/mpir_pmi.c

```diff
@@ -1357,7 +1357,7 @@ static void free_pmi_keyvals(INFO_TYPE ** kv, int size, int *counts)
 
     for (int i = 0; i < size; i++) {
         for (int j = 0; j < counts[i]; j++) {
-            MPL_free(INFO_TYPE_KEY(kv[i][j]));
+            MPL_free((char *) INFO_TYPE_KEY(kv[i][j]));
             MPL_free(INFO_TYPE_VAL(kv[i][j]));
```

Comment:
This next line (`MPL_free(INFO_TYPE_VAL(kv[i][j]));`) is also creating a warning. 
```
src/util/mpir_pmi.c:1360:22: note: in expansion of macro ‘INFO_TYPE_KEY’
 1360 |             MPL_free(INFO_TYPE_KEY(kv[i][j]));
      |                      ^~~~~~~~~~~~~
/tmp/jenkins.tmp.eimW78nY/src/mpl/include/mpl_trmem.h:503:17: note: expected ‘void *’ but argument is of type ‘const char *’
  503 | void MPL_trfree(void *, int, const char[]);
```

Reply:
No. Only key is `const char *` -- https://github.com/pmodels/mpich/blob/fb554b4c154b4d936a9df4483b9a0039cf29276c/src/pmi/include/pmi.h#L420-L423

Reply:
Oops, its the same warning. I misread it - as it looked a little different in few test results.